### PR TITLE
Add field for fixed android appStart time

### PR DESCRIPTION
**Pull request by andrewsavage1**
b/350789128

### CODE REVIEW

File: cobalt/dom/performance_lifecycle_timing.idl [hunk 1]
```diff
@@ -25,4 +25,6 @@ interface PerformanceLifecycleTiming : PerformanceEntry {
   readonly  attribute DOMHighResTimeStamp appDeeplink;
   readonly  attribute DOMString  currentState;
   readonly  attribute DOMString  lastState;
+
```
Comment (kaidokert) [inline cobalt/dom/performance_lifecycle_timing.idl:28]: Please add a comment here and on the `appStar` as well for those being non-spec. 

There should actually be a link to the DOM spec in the header comments here as well, as we do elsewhere in the IDL files.

File: starboard/extension/system_info.h [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  // Name should be the string |kStarboardExtensionSystemInfoName|.
+  // This helps to validate that the extension API is correct.
+  const char* name;
+
+  // This specifies the version of the API that is implemented.
+  uint32_t version;
+
+  // The fields below this point were added in version 1 or later.
+
+  int64_t (*GetAppStartWithAndroidFix)();
```
Comment (kaidokert) [inline starboard/extension/system_info.h:36]: More of a question, not a comment. Do you want to leave it as a very specific API for retrieving this one number ? Or, make it a  bit more generic with 

```
int64_t (*GetSystemMetric)(const char *name)
```
where `name="appstart_with_android_fix"` ? With the latter, we wouldn't need to update the interface  extension version every time someone wants to retrieve a value.

There's as strong YAGNI argument for the latter here, but i also know Media team is trying to do similar things.

Comment (andrewsavage1) [reply starboard/extension/system_info.h:36]: I'd rather leave it very specific with the goal of removing it in a later minor release if possible—it's not a particularly hard thing to implement so I'm not that worried about it potentially being redone

File: starboard/android/shared/application_android.h [hunk 3]
```diff
@@ -95,7 +95,12 @@ class ApplicationAndroid
   void SbWindowSendInputEvent(const char* input_text, bool is_composing);
   void SendLowMemoryEvent();
   void OsNetworkStatusChange(bool became_online);
-  int64_t GetAppStartTimestamp();  // microseconds
+
+  // Returns the appStart time in microseconds. GetIncorrectAppStartTimestamp
+  // exists to report an existing (incorrect) metric.
+  int64_t GetIncorrectAppStartTimestamp();
+  int64_t GetAppStartTimestamp();
+  int64_t app_start_with_android_fix() { return app_start_timestamp_; }
```
[the remark this answers is not in the document]
Comment (andrewsavage1) [reply starboard/android/shared/application_android.h:103]: Kabuki explicitly wants a separate metric for this variable with the android correction, and once they're fine with switching to just the appStart metric then we should remove this plumbing entirely and rely on the start event to send the timestamp to Cobalt.

File: starboard/extension/system_info.h (hunk 2 above)
Comment (kaidokert) [reply starboard/extension/system_info.h:36]: Sgtm.

File: cobalt/dom/performance_lifecycle_timing.idl (hunk 1 above)
Comment (kaidokert) [reply cobalt/dom/performance_lifecycle_timing.idl:28]: Yeah i realized that after i left the comment. We should probably at least leave a comment at the top that it's entirely non-spec. 

Also unsure but there's a related `VisibilityStateEntry` now https://html.spec.whatwg.org/multipage/interaction.html#the-visibilitystateentry-interface

I'm not sure if that existed when we did the original design.

**Pull request merged**